### PR TITLE
feat(zkEVM): adds arithmetic state machine docs

### PR DESCRIPTION
**Pull request by ashucoder9**

### CODE REVIEW

File: docs/zkEVM/zkProver/arithmetic-sm.md [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+#### Remark
+A technicality is that $\text{carry}$ is subdivided into two other $\text{carry}_L$ and $\text{carry}_H$ such that:
+
+$$
+\text{carry} = \text{carry}_L + \text{carry}_H \cdot 2^{18}.
+$$
+
+## Source Code
+
+The Polygon zkEVM Repo is available on [GitHub](https://github.com/0xPolygonHermez).
```
Comment (ashucoder9) [inline docs/zkEVM/zkProver/arithmetic-sm.md:187]:
```suggestion
The Polygon zkEVM repository is available on [GitHub](https://github.com/0xPolygonHermez).
```

File: docs/zkEVM/zkProver/arithmetic-sm.md [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+$$
+\begin{aligned}
+\mathbf{eq\ } &= x_1[0] \cdot y_1[0] \\
+\mathbf{eq'} &= x_1[0] \cdot y_1[1] + x_1[1] \cdot y_1[0]
+\end{aligned}
+$$
+
+But then, the carry generated by $\mathbf{eq}$ has to be taken into account by $\mathbf{eq'}$.
+
+Going back to our equations; $\text{EQ}_0, \text{EQ}_1, \text{EQ}_2, \text{EQ}_4$; let's see how the operation is performed in $\text{EQ}_0$.
```
Comment (cerberushades) [inline docs/zkEVM/zkProver/arithmetic-sm.md:137]:
```suggestion
Going back to our equations $\text{EQ}_0, \text{EQ}_1, \text{EQ}_2, \text{EQ}_4$; let's see how the operation is performed in $\text{EQ}_0$. 
```

File: docs/zkEVM/zkProver/arithmetic-sm.md [hunk 3]
```diff
@@ -0,0 +1,193 @@
...
+Continuing until the last two:
+
+$$
+\begin{aligned}
+\mathbf{eq}_{30} &= (x_1[15] \cdot y_1[15]) - y_2[14], \\
+\mathbf{eq}_{31} &= -y_2[15]. 
+\end{aligned}
+$$
+
+Now, notice that the $\mathbf{eq}_i$'s do not care about the carry they generate. That means, if $\mathbf{eq}_i = 10$, then what we really want the result to be is $0$ and save $1$ as a carry for the next operation. To express this fact as a constraint, we say that the following has to be satisfied:
```
Comment (cerberushades) [inline docs/zkEVM/zkProver/arithmetic-sm.md:172]:
```suggestion
Now, notice that the carry generated by the $\mathbf{eq}_i$'s is not important for them. That means, if $\mathbf{eq}_i = 10$, then what we really want the result to be is $0$ and save $1$ as a carry for the next operation. To express this fact as a constraint, we say that the following has to be satisfied:
```

**Review by cerberushades — CHANGES_REQUESTED** [review]

**Comment by cerberushades** [comment]
@ashucoder9 you've added a few images that weren't used in the article. Is this what you planned?

File: docs/zkEVM/zkProver/arithmetic-sm.md [hunk 4]
```diff
@@ -0,0 +1,191 @@
...
+A \cdot B + C = D \cdot 2^{256} + E \tag{\bf{Eqn A}}
+$$
+
+where $A, B, C, D$ and $E$ are 256-bit integers.
+
+For instance, if $C = 0$, then $\bf{Eqn\ A}$ states that the result of multiplying $A$ and $B$ is $E$ with a carry of $D$. That is, $D$ is the chunk that exceeds 256 bits. 
+
+Or, if $B = 1$,  $\bf{Eqn\ A}$ states that the result of adding $A$ and $C$ is the same as before: $E$ with a carry of $D$. Similarly, division and modular reductions can also be expressed as derivatives of $\bf{Eqn\ A}$.
+
+**These operations are performed in the Arithmetic State Machine, with registers satisfying the following PIL relation**,
```
[the remark this answers is not in the document]
Comment (ashucoder9) [reply docs/zkEVM/zkProver/arithmetic-sm.md:73]: Nope, it's part of the explanation so admonition doesn't seem to fit here.

File: docs/zkEVM/zkProver/arithmetic-sm.md [hunk 5]
```diff
@@ -0,0 +1,191 @@
...
+
+where $q_0,q_1,q_2 \in \mathbb{Z}$, implying that these equations hold true over the integers. 
+
+This approach is taken in order to the need to compute divisions by $p$. Note that **only three possible computation scenarios can arise**:
+
+1. $\text{EQ}_0$ is activated while the rest are deactivated,
+2. $\text{EQ}_1$, $\text{EQ}_3$ and $\text{EQ}_4$ are activated but $\text{EQ}_0$ and $\text{EQ}_2$ are deactivated, 
+3. $\text{EQ}_2$, $\text{EQ}_3$ and $\text{EQ}_4$ are activated and $\text{EQ}_0$ and $\text{EQ}_1$ are deactivated.
+
+Since at most, one of $\text{EQ}_1$ and $\text{EQ}_2$ are activated in any scenario, we can afford "sharing'' the same $q_0$ for both.
```
[the remark this answers is not in the document]
Comment (ashucoder9) [reply docs/zkEVM/zkProver/arithmetic-sm.md:104]: Looks good to me 👀 But please commit if you have some suggestion.

**Comment by ashucoder9** [comment]
> @ashucoder9 you've added a few images that weren't used in the article. Is this what you planned?

Yes, that's intentional for upcoming articles.

**Review by cerberushades — APPROVED** [review]

**Pull request merged**